From 957dca3df624abcbf895f5081fc664693aa0b363 Mon Sep 17 00:00:00 2001
Message-Id: <957dca3df624abcbf895f5081fc664693aa0b363.1648734618.git.geliang.tang@suse.com>
From: Muhammad Usama Anjum <musamaanjum@gmail.com>
Date: Tue, 6 Apr 2021 00:49:04 +0500
Subject: [PATCH] bpf, inode: Remove second initialization of the
 bpf_preload_lock
Git-commit: 957dca3df624abcbf895f5081fc664693aa0b363
Patch-mainline: v5.13-rc1
References: bsn#127
Modified-by-SEL: No


bpf_preload_lock is already defined with DEFINE_MUTEX(). There is no
need to initialize it again. Remove the extraneous initialization.

Signed-off-by: Muhammad Usama Anjum <musamaanjum@gmail.com>
Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>
Link: https://lore.kernel.org/bpf/20210405194904.GA148013@LEGION
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 kernel/bpf/inode.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/kernel/bpf/inode.c b/kernel/bpf/inode.c
index 1576ff331ee4..f441d521ef77 100644
--- a/kernel/bpf/inode.c
+++ b/kernel/bpf/inode.c
@@ -816,8 +816,6 @@ static int __init bpf_init(void)
 {
 	int ret;
 
-	mutex_init(&bpf_preload_lock);
-
 	ret = sysfs_create_mount_point(fs_kobj, "bpf");
 	if (ret)
 		return ret;
-- 
2.34.1

